linera-core/src/client/mod.rs

```diff
@@ -3022,13 +3022,19 @@ where
             let hash_map::Entry::Vacant(entry) = senders.entry(name) else {
                 continue;
             };
-            let (mut stream, abort) = match node.subscribe(vec![chain_id]).await {
-                Err(error) => {
+            let stream = stream::once({
+                let node = node.clone();
+                async move { node.subscribe(vec![chain_id]).await }
+            })
+            .filter_map(move |result| async move {
```

Comment:
This `async move` is a bit unnecessary, isn't it? i.e. the `if let` doesn't need the `async { ... }` wrapper but it may unnecessarily allocate the future?

Reply:
But `filter_map` needs a future that outputs an option, not an option of a future.